src/legacy/core_plugins/vis_type_timelion/public/components/panel.tsx

```diff
@@ -0,0 +1,361 @@
+      caption.html(emptyCaption);
+      setLegendCaption(caption);
+
+      const canvasNode = $(canvasElem);
+      canvasNode.find('div.legend table').append(caption);
+      setLegendValueNumbers(canvasNode.find('.ngLegendValueNumber'));
+
+      // legend has been re-created. Apply focus on legend element when previously set
+      if (focusedSeries || focusedSeries === 0) {
+        canvasNode
+          .find('div.legend table .legendLabel>span')
+          .get(focusedSeries)
+          .focus();
+      }
+    }
+  }, [plot, focusedSeries, canvasElem]);
+
+  const highlightSeries = useCallback(
+    debounce(({ currentTarget }: JQuery.TriggeredEvent) => {
+      const id = Number(currentTarget.getAttribute(SERIES_ID_ATTR));
+      if (highlightedSeries === id) {
+        return;
+      }
+
+      setHighlightedSeries(id);
+      setChart(chartState =>
+        chartState.map((series: Series, seriesIndex: number) => {
+          const color =
+            seriesIndex === id
+              ? originalColorMap.get(series) // color it like it was
+              : 'rgba(128,128,128,0.1)'; // mark as grey
+
+          return { ...series, color };
+        })
+      );
+    }, DEBOUNCE_DELAY),
+    [highlightedSeries, originalColorMap]
+  );
+
+  const focusSeries = useCallback(
+    (event: JQuery.TriggeredEvent) => {
+      const id = Number(event.currentTarget.getAttribute(SERIES_ID_ATTR));
+      setFocusedSeries(id);
+      highlightSeries(event);
+    },
+    [highlightSeries]
+  );
+
+  const toggleSeries = useCallback(({ currentTarget }: JQuery.TriggeredEvent) => {
+    const id = Number(currentTarget.getAttribute(SERIES_ID_ATTR));
+    setChart(chartState =>
+      chartState.map((series: Series, seriesIndex: number) => {
+        return seriesIndex === id ? { ...series, _hide: !series._hide } : { ...series };
+      })
+    );
+  }, []);
+
+  const options = useMemo(
+    () =>
+      buildOptions(
+        interval,
+        kibana.services.timefilter,
+        kibana.services.uiSettings,
+        chartElem && chartElem.clientWidth,
+        seriesList.render.grid
+      ),
+    [seriesList.render.grid, interval, chartElem, kibana.services]
+  );
+
+  const updatedSeries = useMemo(() => buildSeriesData(chart, options), [chart, options]);
+
+  useEffect(() => {
+    if (canvasElem && canvasElem.clientWidth > 0 && canvasElem.clientHeight > 0) {
+      // @ts-ignore
+      setPlot($.plot(canvasElem, compact(updatedSeries), options));
+      renderComplete();
+
+      const legend = $(canvasElem).find('.ngLegendValue');
+      if (legend) {
+        legend.click(toggleSeries);
+        legend.focus(focusSeries);
+        legend.mouseover(highlightSeries);
+      }
+    }
+  }, [
+    canvasElem,
+    options,
+    updatedSeries,
+    toggleSeries,
+    focusSeries,
+    highlightSeries,
+    renderComplete,
+  ]);
+
+  useEffect(() => {
+    moment.tz.setDefault(kibana.services.uiSettings.get('dateFormat:tz'));
```

Comment:
It looks like `src/core/public/integrations/moment/moment_service.ts` is already doing the same thing, so I don't think it's necessary here.